packages/flutter_tools/lib/src/reporting/usage.dart

```diff
@@ -187,13 +214,21 @@ class _DefaultUsage implements Usage {
     if (usingLogFile) {
       _analytics = LogToFileAnalytics(logFilePath);
     } else {
-      _analytics = AnalyticsIO(
-            _kFlutterUA,
-            settingsName,
-            version,
-            documentDirectory:
-                configDirOverride != null ? globals.fs.directory(configDirOverride) : null,
-          );
+      try {
+        _analytics = analyticsIOFactory(
```

Comment:
I think this ends up as `null` if no value is provided, you might need to do `analyticsIOFactory ?? _default...`

Reply:
Because the `Usage` factory constructor unconditionally passes `analyticsIOFactory`

Reply:
Oh, good catch! Fixed.